### PR TITLE
LDAP login support

### PR DESCRIPTION
**Pull request by alisnic**

### CODE REVIEW

**Comment by frenkel** [comment]
Looks good, thank you. Would a user need to modify the code in the User model to switch to ldap?

**Comment by alisnic** [comment]
Yes, that is correct.

**Comment by frenkel** [comment]
Would it be possible to make it configurable? I mean something like: sign in with a "database account" configure your ldap settings, save it and use ldap accounts from then on (or combined with "database accounts").

**Comment by alisnic** [comment]
If I understand you correctly, you want to switch the authentication strategy for the User model on the fly. I'm not sure this is a good idea, I need to investigate. But I can make the strategy configurable by putting it as a field in Rails application config. (if you are trying to avoid editing the user model to change the auth strategy)

thoughts?

**Comment by frenkel** [comment]
For our hosted Brimir solution it would be great to have it switchable on the fly, because certain user could then configure it themselves, while other use `:database_authentication`. If it's not possible, a Rails configuration option might be better indeed, because it makes upgrades easier.

**Comment by alisnic** [comment]
I've updated the pull request, check if I understood you correctly. In regards to on-the-fly switching of auth strategies - that's not easy nor beautiful in my opinion, so it's not something I'm interested in doing right now (although I think it's possible to do it). I'll be more than happy to polish and tweak the current implementation to meet your standards.

**Comment by frenkel** [comment]
Thanks, I've added some comments

**Comment by alisnic** [comment]
I have updated the README, sorry about the silly typos

File: Gemfile [hunk 1]
```diff
@@ -40,6 +40,7 @@ gem 'sqlite3', group: :sqlite
 
 # authentication
 gem 'devise'
+gem "devise_ldap_authenticatable"
```
Comment (frenkel) [inline Gemfile]: No need to use double quotes here. Could you use single quotes please?

**Comment by frenkel** [comment]
Final last, tiny comment added. If you fix that I'll merge it. Thanks again!

**Comment by alisnic** [comment]
Done.

**Comment by frenkel** [comment]
Merged! Thanks a lot.

**Comment by alisnic** [comment]
Cool! Thank you for making this awesome project open!

**Pull request closed without merging**